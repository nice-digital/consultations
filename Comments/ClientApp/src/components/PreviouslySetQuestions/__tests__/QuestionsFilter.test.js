import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { QuestionsFilter } from "../QuestionsFilter";

it("should fire the handleFilter function when a input is changed", () => {
	const fakeProps = {
		handleFilter: jest.fn(),
		filter: {
			direcorate: true,
		},
	};
	render(<QuestionsFilter {...fakeProps}/>);
	const radio1 = screen.getByLabelText("All Directorates");
	fireEvent.click(radio1);
	expect(fakeProps.handleFilter).toHaveBeenCalled();
});

test("should match the snapshot when the filter is set to directorate", () => {
	const fakeProps = {
		handleFilter: jest.fn(),
		filter: {
			direcorate: true,
		},
	};
	const {container} = render(<QuestionsFilter {...fakeProps} />);
	expect(container).toMatchSnapshot();
});

test("should match the snapshot when the filter is set to all", () => {
	const fakeProps = {
		handleFilter: jest.fn(),
		filter: {
			direcorate: false,
		},
	};
	const {container} = render(<QuestionsFilter {...fakeProps} />);
	expect(container).toMatchSnapshot();
});
