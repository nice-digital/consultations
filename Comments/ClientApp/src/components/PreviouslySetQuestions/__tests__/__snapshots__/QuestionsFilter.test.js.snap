// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot when the filter is set to all 1`] = `
<div>
  <div
    class="panel mb--d"
  >
    <div
      class="form__group form__group--text"
    >
      <label
        class="form__label"
        for="textFilter"
      >
        <b>
          Filter by question text
        </b>
      </label>
      <input
        class="form__input"
        id="textFilter"
        tabindex="0"
        value=""
      />
    </div>
    <div
      role="radiogroup"
    >
      <label
        class="form__label"
        for="filterByRole"
      >
        <b>
          Filter by directorate
        </b>
      </label>
      <div
        class="form__group form__group--radio form__group--inline"
      >
        <input
          class="form__radio"
          id="filterByRole--mine"
          name="filterByRole"
          type="radio"
          value="true"
        />
        <label
          class="form__label form__label--radio"
          for="filterByRole--mine"
        >
          My Directorate
        </label>
      </div>
      <div
        class="form__group form__group--radio form__group--inline"
      >
        <input
          class="form__radio"
          id="filterByRole--all"
          name="filterByRole"
          type="radio"
          value="false"
        />
        <label
          class="form__label form__label--radio"
          for="filterByRole--all"
        >
          All Directorates
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`should match the snapshot when the filter is set to directorate 1`] = `
<div>
  <div
    class="panel mb--d"
  >
    <div
      class="form__group form__group--text"
    >
      <label
        class="form__label"
        for="textFilter"
      >
        <b>
          Filter by question text
        </b>
      </label>
      <input
        class="form__input"
        id="textFilter"
        tabindex="0"
        value=""
      />
    </div>
    <div
      role="radiogroup"
    >
      <label
        class="form__label"
        for="filterByRole"
      >
        <b>
          Filter by directorate
        </b>
      </label>
      <div
        class="form__group form__group--radio form__group--inline"
      >
        <input
          class="form__radio"
          id="filterByRole--mine"
          name="filterByRole"
          type="radio"
          value="true"
        />
        <label
          class="form__label form__label--radio"
          for="filterByRole--mine"
        >
          My Directorate
        </label>
      </div>
      <div
        class="form__group form__group--radio form__group--inline"
      >
        <input
          class="form__radio"
          id="filterByRole--all"
          name="filterByRole"
          type="radio"
          value="false"
        />
        <label
          class="form__label form__label--radio"
          for="filterByRole--all"
        >
          All Directorates
        </label>
      </div>
    </div>
  </div>
</div>
`;
